src/Avalonia.Diagnostics/Diagnostics/ViewModels/PropertyViewModel.cs

```diff
@@ -25,19 +26,37 @@ protected static string ConvertToString(object value)
                 return "(null)";
             }
 
-            var converter = TypeDescriptor.GetConverter(value);
-            return converter?.ConvertToString(value) ?? value.ToString();
+            //Check if there's an user provided ToString(), prefer that over the TypeDescriptor conversion
+            if (value.GetType().GetMethod(nameof(ToString), System.Type.EmptyTypes)
+                .DeclaringType != typeof(object))
+            {
+                return value.ToString();
+            }
+
+            try
+            {
+                var converter = TypeDescriptor.GetConverter(value);
```

Comment:
Note that `GetConverter()` never returns null, but rather throws or returns a default converter.